module/documents/advancement/item-choice.mjs

```diff
@@ -0,0 +1,120 @@
+    return this.value.added?.[level] !== undefined;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  titleForLevel(level, { configMode=false }={}) {
+    return `${this.title} <em>(${game.i18n.localize("DND5E.AdvancementChoices")})</em>`;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  summaryForLevel(level, { configMode=false }={}) {
+    const items = this.value.added?.[level];
+    if ( !items || configMode ) return "";
+    return Object.values(items).reduce((html, uuid) => html + game.dnd5e.utils.linkForUuid(uuid), "");
+  }
+
+  /* -------------------------------------------- */
+  /*  Application Methods                         */
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  storagePath(level) {
+    return `value.added.${level}`;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Verify that the provided item can be used with this advancement based on the configuration.
+   * @param {Item5e} item                  Item that needs to be tested.
+   * @param {object} config
+   * @param {string} config.type           Type restriction on this advancement.
+   * @param {object} config.restriction    Additional restrictions to be applied.
+   * @param {boolean} [config.error=true]  Should an error be thrown when an invalid type is encountered?
+   * @returns {boolean}                    Is this type valid?
+   * @throws An error if the item is invalid and warn is `true`.
+   */
+  _validateItemType(item, { type, restriction, error=true }={}) {
+    super._validateItemType(item, { error });
+    type ??= this.configuration.type;
+    restriction ??= this.configuration.restriction;
+
+    // Type restriction is set and the item type does not match the selected type
+    if ( type && (type !== item.type) ) {
+      const typeLabel = game.i18n.localize(`ITEM.Type${restriction.capitalize()}`);
+      if ( error ) throw new Error(game.i18n.format("DND5E.AdvancementItemChoiceTypeWarning", { type: typeLabel }));
+      return false;
+    }
+
+    // If additional type restrictions applied, make sure they are valid
+    if ( (type === "feat") && restriction.type ) {
+      const typeConfig = CONFIG.DND5E.featureTypes[restriction.type];
+      const subtype = typeConfig.subtypes?.[restriction.subtype];
+      let errorLabel;
+      if ( restriction.type !== item.system.type.value ) errorLabel = typeConfig.label;
+      else if ( subtype && (restriction.subtype !== item.system.type.subtype) ) errorLabel = subtype;
+      if ( errorLabel ) {
+        if ( error ) throw new Error(game.i18n.format("DND5E.AdvancementItemChoiceTypeWarning", { type: errorLabel }));
+        return false;
+      }
+    }
+
+    // If spell level is restricted, ensure the spell is of the appropriate level
+    const l = parseInt(restriction.level);
+    if ( (type === "spell") && Number.isNumeric(l) && (item.system.level !== l) ) {
```

Comment:
We can do `Number.isNumeric(restriction.level)` first to check if the string contains only digits, then do `parseInt(restriction.level)` inside the if-statement. If we do the `parseInt` first, and it's not a number, we'll get a `NaN`, which will still correctly return `false` from `Number.isNumeric`, but feels like the wrong way around to me. Either that or we check for `isNaN(l)` rather than using `Number.isNumeric`, which is supposed to be used on strings.

Reply:
I'll use `isNaN` to keep the line from going beyond our maximum line length :).